include/core/density_model.h

```diff
@@ -51,16 +41,58 @@ class DensityConstant : public DensityModel
   {}
 
   /**
-   * @brief Returns the denisty
-   *
-   * @param temperature Temperature at time t+dt
+   * @brief value Calculates the value of a physical property.
+   * @param fields_value Value of the various field on which the property may depend.
+   * @return value of the physical property calculated with the fields_value
    */
-  virtual inline double
-  get_density(const double /*temperature*/) override
+  virtual double
+  value(const std::map<field, double> /*fields_value*/)
   {
     return density;
+  };
+
+  /**
+   * @brief vector_value Calculates the values of a physical property for
+   * @param field_vectors
+   */
+  virtual void
+  vector_value(const std::map<field, std::vector<double>> & /*field_vectors*/,
+               std::vector<double> &property_vector)
+  {
+    property_vector.assign(property_vector.size(), density);
```

Comment:
I think the fastest way to fill a std::vector with te same value is to use this :
std::fill(property_vector.begin(), property_vector.end(), dentity);
But of course I'm not sure about this.

Reply:
Ouhh nice, did not see that. I'll switch to that I think